From bf611fbbf3183e6f96529d79189be56189a46e1b Mon Sep 17 00:00:00 2001
From: vaxerski <vaxry@vaxry.net>
Date: Sun, 18 Aug 2024 22:40:21 +0200
Subject: [PATCH 04/20] screencopy: nuke unused stuff

---
 src/protocols/Screencopy.cpp | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/protocols/Screencopy.cpp b/src/protocols/Screencopy.cpp
index f246f6dd..8a7fd567 100644
--- a/src/protocols/Screencopy.cpp
+++ b/src/protocols/Screencopy.cpp
@@ -38,17 +38,6 @@ CScreencopyFrame::CScreencopyFrame(SP<CZwlrScreencopyFrameV1> resource_, int32_t
 
     g_pHyprRenderer->makeEGLCurrent();
 
-    if (g_pHyprOpenGL->m_mMonitorRenderResources.contains(pMonitor)) {
-        const auto& RDATA = g_pHyprOpenGL->m_mMonitorRenderResources.at(pMonitor);
-        // bind the fb for its format. Suppress gl errors.
-#ifndef GLES2
-        glBindFramebuffer(GL_READ_FRAMEBUFFER, RDATA.offloadFB.m_iFb);
-#else
-        glBindFramebuffer(GL_FRAMEBUFFER, RDATA.offloadFB.m_iFb);
-#endif
-    } else
-        LOGM(ERR, "No RDATA in screencopy???");
-
     shmFormat = g_pHyprOpenGL->getPreferredReadFormat(pMonitor);
     if (shmFormat == DRM_FORMAT_INVALID) {
         LOGM(ERR, "No format supported by renderer in capture output");
-- 
2.45.2

