From f4045ab8d032186b7c5409cd9f05eeaa813320ba Mon Sep 17 00:00:00 2001
From: vaxerski <vaxry@vaxry.net>
Date: Sun, 18 Aug 2024 22:57:21 +0200
Subject: [PATCH 06/20] screencopy: fix 10b format r/b flip

---
 src/protocols/Screencopy.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/protocols/Screencopy.cpp b/src/protocols/Screencopy.cpp
index 8a7fd567..b25d9456 100644
--- a/src/protocols/Screencopy.cpp
+++ b/src/protocols/Screencopy.cpp
@@ -46,6 +46,10 @@ CScreencopyFrame::CScreencopyFrame(SP<CZwlrScreencopyFrameV1> resource_, int32_t
         return;
     }
 
+    // TODO: hack, we can't bit flip so we'll format flip heh, GL_BGRA_EXT wont work here
+    if (shmFormat == DRM_FORMAT_XRGB2101010 || shmFormat == DRM_FORMAT_ARGB2101010)
+        shmFormat = DRM_FORMAT_XBGR2101010;
+
     const auto PSHMINFO = FormatUtils::getPixelFormatFromDRM(shmFormat);
     if (!PSHMINFO) {
         LOGM(ERR, "No pixel format supported by renderer in capture output");
-- 
2.45.2

