From 1b1ecf77e0c195460eb5335652d65de6fd83cf7b Mon Sep 17 00:00:00 2001
From: Mihai Fufezan <mihai@fufexan.net>
Date: Wed, 21 Aug 2024 22:37:28 +0300
Subject: [PATCH 17/20] Nix: include xcursor regardless of xwayland

---
 nix/default.nix | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nix/default.nix b/nix/default.nix
index 9bae9d83..c8eaf731 100644
--- a/nix/default.nix
+++ b/nix/default.nix
@@ -136,11 +136,11 @@ assert lib.assertMsg (!hidpiXWayland) "The option `hidpiXWayland` has been remov
         tomlplusplus
         wayland
         wayland-protocols
+        xorg.libXcursor
       ]
       (lib.optionals stdenv.hostPlatform.isMusl [libexecinfo])
       (lib.optionals enableXWayland [
         xorg.libxcb
-        xorg.libXcursor
         xorg.libXdmcp
         xorg.xcbutil
         xorg.xcbutilerrors
-- 
2.45.2

