From 8162fae37728b3ae2b1a82a32663e76c59c4ce61 Mon Sep 17 00:00:00 2001
From: James R Larrowe <larrowe.semaj11@gmail.com>
Date: Tue, 20 Aug 2024 13:54:51 -0400
Subject: [PATCH 19/20] Fix Makefile too

... did this ever work?
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index adf6fbe8..bc7b750b 100644
--- a/Makefile
+++ b/Makefile
@@ -87,8 +87,8 @@ asan:
 	#git reset --hard
 
 	@echo -en "If you want to apply a patch, input its path (leave empty for none):\n"
-	@read patchvar
-	@if [-n "$patchvar"]; then patch -p1 < $patchvar || echo ""; else echo "No patch specified"; fi
+	@read patchvar; \
+	 if [ -n "$$patchvar" ]; then patch -p1 < "$$patchvar" || echo ""; else echo "No patch specified"; fi
 
 	git clone --recursive https://gitlab.freedesktop.org/wayland/wayland
 	cd wayland && patch -p1 < ../scripts/waylandStatic.diff && meson setup build --buildtype=debug -Db_sanitize=address -Ddocumentation=false && ninja -C build && cd ..
-- 
2.45.2

